Minor build improvements

kgaillot opened the pull request:
Most significantly, this moves the various rpm-related make targets from GNUmakefile to a new rpm/Makefile.am, and adds a configure --with-corosync-conf option to specify the location of the Corosync configuration file (normally /etc/corosync/corosync.conf)

clumens commented on GNUmakefile:
```diff
@@ -17,6 +17,9 @@ EXTRA_CLEAN_TARGETS	= ancillary-clean
 
+# Define release-related variables
+include $(top_srcdir)/mk/release.mk
+
```
If autogen.sh and configure haven't been run first, won't this mean $(top_srcdir) will be undefined and this will try to include /mk/release.mk?

kgaillot replied:
whoops good catch -- i can default top_srcdir to .

clumens commented on doc/Makefile.am:
```diff
@@ -68,8 +68,10 @@ deprecated-clean:
 global:
 	$(MAKE) $(AM_MAKEFLAGS) -C .. clean-generic
+	$(MAKE) $(AM_MAKEFLAGS) -C ../rpm rpm-clean
 	cd .. && gtags -q && htags -sanhIT doc
```
It might be worth adding the global package as an optional build dependency in INSTALL.md, then.  I didn't know about this until just now.

kgaillot commented:
Updated "define release variables in include file" commit to use abs_srcdir (which already has a default), and added "mention global as optional build dependency" commit

Not sure how to reword the make/getopt thing

kgaillot commented:
> > Hard to tell how important that is. Let me play a bit with what we have atm (with this PR).
> > In theory it should be possible as well to have some targets in Makefile.am formulated that they can be understood by make directly as done e.g. with sbd. Haven't checked what that would mean for pacemaker though.
> 
> That might work if I add defaults for top_srcdir, abs_builddir, and PACKAGE in rpm/Makefile.am

@wenningerk , added a commit on top to add those defaults, but I haven't tested it

kgaillot commented on INSTALL.md:
```diff
@@ -17,7 +17,8 @@
 
-Also: GNU make
+Also, make must be GNU (or compatible), and GNU (or compatible) getopt
+must be somewhere on the PATH.
```
That looks good -- rebased on current master and updated that commit

wenningerk commented:
> > > Hard to tell how important that is. Let me play a bit with what we have atm (with this PR).
> > > In theory it should be possible as well to have some targets in Makefile.am formulated that they can be understood by make directly as done e.g. with sbd. Haven't checked what that would mean for pacemaker though.
> > 
> > 
> > That might work if I add defaults for top_srcdir, abs_builddir, and PACKAGE in rpm/Makefile.am
> 
> @wenningerk , added a commit on top to add those defaults, but I haven't tested it

```make``` called from the rpm subdir (except the ```BUILD_HELP``` thing mentioned below) gives me a src-rpm tainting the git-tree.
Didn't work for me out of the box trying to build out of tree as not to taint git.
Thus tried a small modification.
```
--- a/rpm/Makefile.am
+++ b/rpm/Makefile.am
@@ -17,8 +17,6 @@ PACKAGE               ?= pacemaker
 AM_V_at                ?= @
 MKDIR_P                ?= mkdir -p
 
-include $(top_srcdir)/mk/common.mk
-
 # Define release-related variables
 include $(top_srcdir)/mk/release.mk
 
@@ -51,6 +49,16 @@ RPM_CLEAN_subtree    = "$(abs_builddir)/BUILD"       \
                          "$(abs_builddir)/SPECS"       \
                          "$(abs_builddir)/SRPMS"
 
+RPM_SPEC_DIR_flat      = $(abs_builddir)
+RPM_SRCRPM_DIR_flat    = $(abs_builddir)
+RPM_OPTS_flat          = --define "_sourcedir $(abs_builddir)"              \
+                          --define "_topdir $(abs_builddir)"
+RPM_CLEAN_flat= "$(abs_builddir)"       \
+                          "$(abs_builddir)/BUILDROOT"   \
+                          "$(abs_builddir)/RPMS"        \
+                          "$(abs_builddir)/SPECS"       \
+                          "$(abs_builddir)/SRPMS"
+
 RPM_SPEC_DIR   = $(RPM_SPEC_DIR_$(RPMDEST))
 RPM_SRCRPM_DIR = $(RPM_SRCRPM_DIR_$(RPMDEST))
 RPM_OPTS       = $(RPM_OPTS_$(RPMDEST))
```

That allows me to do the following from the directory from where I had done the ```git clone```
with ```DISTRO``` set to the mock config.

```
mkdir -p ${DISTRO}
rm -rf ${DISTRO}/*.tar.gz ${DISTRO}/rpm
make RPMDEST="flat" DIRTY="yes" abs_builddir=$PWD/${DISTRO}/rpm -C ${PACKAGE}/rpm -f Makefile.am srpm
mock --with=doc --with=hardening --with=cibsecrets --without=pre_release --no-clean --root ${DISTRO} --disable-plugin=selinux --disable-plugin=yum_cache --resultdir=${PWD}/${DISTRO} ${DISTRO}/rpm/SRPMS/${PACKAGE}*.src.rpm
```

The ```Makefile.am``` where the rpm-stuff had been before didn't pull in ```common.mk``` either.
Problem was ```if BUILD_HELP``` that ```make``` doesn't understand and - with out of tree build - ```rpmbuild``` not finding the tar-file.
Didn't play - and never had done before - with the mock-targets yet.

kgaillot commented:
OK, I think I have a good solution. I've made substantial changes to some of the commits:

* e672959 (define release variables in include file) moves GNUmakefile's distdir definition to release.mk as top_distdir
* 4dacb12  (separate man page builders from common.mk) is new
* d8dedda (move rpm- and mock-related make targets) moves GNUmakefile's TARFILE variable and export target to rpm/Makefile.am
* eef5bde (allow specifying an arbitrary directory for RPM artifacts) is new

@wenningerk , can you review those and give it another test?

kgaillot commented:
> Alternative approach would be resetting DIRTY if there is nothing to commit and fall back to building HEAD - which would have the benefit of not adding the `.mod` even when there is nothing modified.

Added two commits on top for this
